# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/base/strings/utf_string_conversion_utils.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/base/strings/utf_string_conversion_utils.cc
--- a/security/sandbox/chromium/base/strings/utf_string_conversion_utils.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/base/strings/utf_string_conversion_utils.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -5,6 +5,7 @@
 #include "base/strings/utf_string_conversion_utils.h"
 
 #include "base/third_party/icu/icu_utf.h"
+#include "build/build_config.h"
 
 namespace base {
 
@@ -121,11 +122,11 @@
 }
 
 // Instantiate versions we know callers will need.
+#if !defined(OS_WIN)
+// wchar_t and char16 are the same thing on Windows.
 template void PrepareForUTF8Output(const wchar_t*, size_t, std::string*);
-#ifndef __MINGW32__
-// On MinGW, these instantiations are the same, and MinGW complains if there are two
+#endif
 template void PrepareForUTF8Output(const char16*, size_t, std::string*);
-#endif
 
 template<typename STRING>
 void PrepareForUTF16Or32Output(const char* src,
@@ -145,10 +146,10 @@
 }
 
 // Instantiate versions we know callers will need.
+#if !defined(OS_WIN)
+// std::wstring and string16 are the same thing on Windows.
 template void PrepareForUTF16Or32Output(const char*, size_t, std::wstring*);
-#ifndef __MINGW32__
-// On MinGW, these instantiations are the same, and MinGW complains if there are two
+#endif
 template void PrepareForUTF16Or32Output(const char*, size_t, string16*);
-#endif
 
 }  // namespace base